tests/PHPStan/Parser/TypeParserTest.php

```diff
@@ -276,6 +277,66 @@ public function provideParseData(): array
 				]),
 				Lexer::TOKEN_INTERSECTION,
 			],
+			[
+				'string~int',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'string ~ int',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'(string ~ int)',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'(' . PHP_EOL .
+				'  string' . PHP_EOL .
+				'  ~' . PHP_EOL .
+				'  int' . PHP_EOL .
+				')',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'string~int~float',
```

Comment:
One non-parentheised expression should not allow more than one `~` to prevent associativity ambiguity as not standartized not needed.

As you can see, currently `float` is even missing in the parsed type.